presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -125,6 +138,14 @@ public Block readBlock(ColumnDescriptor columnDescriptor, Type type)
         return columnReader.readBlock(type);
     }
 
+    private byte[] allocateBlock(int length)
+    {
+        byte[] buffer = new byte[length];
+        LocalMemoryContext blockMemoryContext = currentRowGroupMemoryContext.newLocalMemoryContext();
+        blockMemoryContext.setBytes(buffer.length);
```

Comment:
I was thinking like, each time `advanceToNextRowGroup()` is called a new row group gets allocated and that's accounted for in the `blockMemoryContext` (each row group causes a large `buffer` allocation) and it gets cleared with the next `advanceToNextRowGroup()` call. So the memory usage of the reader will look like a pulse going up and down as row groups get allocated/deallocated.  The `buffer` is effectively held by the reader until the next `advanceToNextRowGroup()` call.

What's the right way to capture this?